server/src/test/java/org/elasticsearch/cluster/routing/allocation/InSyncAllocationIdTests.java

```diff
@@ -107,7 +110,7 @@ public void testInSyncAllocationIdsUpdated() {
         clusterState = allocation.disassociateDeadNodes(clusterState, true, "reroute");
 
         // in-sync allocation ids should not be updated
-        assertThat(shardsWithState(clusterState.getRoutingNodes(), UNASSIGNED).size(), equalTo(3));
+        assertThat(shardsWithState(clusterState.getRoutingNodes(), UNASSIGNED).size(), equalTo(6));
```

Comment:
How that even had something allocated if all nodes were gone?

Reply:
It didn't - the shards of `test-old` weren't even in the routing table before.